instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -0,0 +1,325 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing
+
+import httpcore
+import httpx
+import wrapt
+
+from opentelemetry import context
+from opentelemetry.instrumentation.httpx.version import __version__
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.utils import (
+    http_status_to_status_code,
+    unwrap,
+)
+from opentelemetry.propagate import inject
+from opentelemetry.trace import SpanKind, Tracer, TracerProvider, get_tracer
+from opentelemetry.trace.span import Span
+from opentelemetry.trace.status import Status
+
+ResponseInfo = typing.Tuple[
+    int, typing.List[typing.Tuple[bytes, bytes]], typing.Iterable[bytes], dict,
+]
+NameCallback = typing.Callable[[str, str], str]
+SpanCallback = typing.Callable[[Span, ResponseInfo], None]
+URL = typing.Tuple[bytes, bytes, typing.Optional[int], bytes]
+Headers = typing.List[typing.Tuple[bytes, bytes]]
+
+
+def _get_tracer(
```

Comment:
You should be calling get_tracer only once and storing the result no the instrumentor instance or passing it around to different functions. Creating a new tracer for every span is unnecessary overhead no matter how small. This function wouldn't be needed if you used same tracer instance.

Reply:
This is removed now.